hypershift-operator/controllers/machineconfigserver/machineconfigserver_controller.go

```diff
@@ -251,20 +238,23 @@ func (r *MachineConfigServerReconciler) reconcileMCSServiceNodePort(ctx context.
 				TargetPort: intstr.FromInt(8080),
 			},
 		}
-		if serviceNodePort > 0 {
-			mcsService.Spec.Ports[0].NodePort = serviceNodePort
-		}
 		mcsService.Spec.Selector = map[string]string{
 			"app": fmt.Sprintf("machine-config-server-%s", mcs.Name),
 		}
 		mcsService.Spec.Type = corev1.ServiceTypeNodePort
+
+		// If there's user input nodePort and there's no existing one, we set it.
+		if mcs.Spec.IgnitionService.NodePort.Port > 0 && !(mcsService.Spec.Ports[0].NodePort > 0) {
+			mcsService.Spec.Ports[0].NodePort = mcs.Spec.IgnitionService.NodePort.Port
+		}
```

Comment:
If these values diverge, will this code update the existing port to a new value? If so I think that will that cause an error unless transitioning also to ExternalName.

Reply:
if there's already a port set, it won't be overridden since `!(mcsService.Spec.Ports[0].NodePort > 0)` won't be met, agree?

Reply:
Oh, yeah, you're right. Would this be equivalent but easier to follow at a glance?

```go
mcs.Spec.IgnitionService.NodePort.Port > 0 && mcsService.Spec.Ports[0].NodePort == 0
```

Reply:
sgtm, changed to `<= 0`